### PR TITLE
Make `slds-col` optional for Columns & remove deprecated flavors, closes #103

### LINKED ISSUE #103: `Column` does not support omitting `slds-col`
See: https://github.com/propertybase/react-lds/blob/master/src/components/Grid/Column.js#L24
See: https://lightningdesignsystem.com/components/utilities/grid/#flavor-automatic-sizing-no-stretch

- [ ] Column should set slds-col by default, but have the option to not set it
- [ ] Remove deprecated flavors, see: https://lightningdesignsystem.com/components/utilities/grid/#overview
- [ ] Check if there are new flavors
- [ ] Update `pba-react` components that use deprecated flavors or could benefit from the improved `Column`

### PR DESCRIPTION
**Pull request by tomlord90**

### CODE REVIEW

File: src/components/Grid/Column.js [hunk 1]
```diff
@@ -17,13 +17,14 @@ const breakPointProp = breakpoint => `${breakpoint}-sizeOf`;
 const sizeRegex = /^([1-9]|1[0-2])-([1-9]|1[0-2])$/;
 
 export const Column = (props, { cssPrefix }) => {
-  const { align, children, className, ...rest } = props;
+  const { align, children, className, omitSldsCol, ...rest } = props;
```
Comment (fspoettel) [inline src/components/Grid/Column.js]: `omitCol` is sufficient here

File: src/components/Grid/Column.js [hunk 2]
```diff
@@ -17,13 +17,14 @@ const breakPointProp = breakpoint => `${breakpoint}-sizeOf`;
...
+
   const prefix = (classes, passThrough) => prefixClasses(cssPrefix, classes, passThrough);
 
-  const sldsClasses = [
-    'col',
-    { [`align-${align}`]: !!align },
-  ];
+  const sldsClasses = [];
```
Comment (fspoettel) [inline src/components/Grid/Column.js]:
``` js
const sldsClasses = [
  { col: !omitCol },
  { [`align-${align}`]: !!align }
];
```
We are using the `classNames` library here, in order to not having to manipulate the class array with array methods. See above code for an example of usage.

**Comment by fspoettel** [comment]
Nice, this is almost good-to-go. Please write a unit test for the behavior. Check the other specs or the enzyme/jest documentation for information on how to do this.

**Comment by PeterSiegl** [comment]
Please squash commits then 👍

**Pull request merged**